### PR TITLE
chore: correct regex in align-version script

### PR DESCRIPTION
**Pull request by jogold**
### Commit Message

The current regex prevents from using `0.0.0` anywhere in the `package.json`.
Enclose with quotes to be more specific and correctly escape dots.

See https://github.com/aws/aws-cdk/pull/7169#issuecomment-609821470 for a failing
build using `0.0.0` in `cdk-build > pre`.

### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: scripts/align-version.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ ${scriptdir}/align-version.js ${files}
 
 # validation
 marker=$(node -p "require('./scripts/get-version-marker')")
-if find . -name package.json | grep -v node_modules | xargs grep "[^0-9]${marker}"; then
+if find . -name package.json | grep -v node_modules | xargs grep -P "\"[^0-9]?${marker}\""; then
```
Comment (RomainMuller) [inline scripts/align-version.sh:17]: I wanted to suggest something like that on initial review of that regex.

It should however catch *all* kinds of quotes (backticks, single, double - minimum).

Comment (RomainMuller) [reply scripts/align-version.sh:17]: Alternatively, we could ignore quotes and exclude an additional `.0` that'd make it into an IP address.

**Comment by RomainMuller** [comment]
As a secondary note - I wonder if we should he a `-pre` suffix for our version token to reduce the likelihood of conflict with a "real" thing.

File: scripts/align-version.sh (hunk 1 above)
Comment (jogold) [reply scripts/align-version.sh:17]: in fact when switching to regex here (`-P`) we have a problem with the dots (`.`) coming from the `0.0.0` (exported from js) not being escaped

Comment (jogold) [reply scripts/align-version.sh:17]: and it should be `-E`

Comment (jogold) [reply scripts/align-version.sh]: @RomainMuller why would you want to check for all kinds of quotes, it's a find in package.json files and other quotes are not valid json, no?

File: scripts/align-version.sh [hunk 2]
```diff
@@ -13,8 +13,8 @@ files="$(find . -name package.json | grep -v node_modules | xargs)"
 ${scriptdir}/align-version.js ${files}
 
 # validation
-marker=$(node -p "require('./scripts/get-version-marker')")
-if find . -name package.json | grep -v node_modules | xargs grep "[^0-9]${marker}"; then
+marker=$(node -p "require('./scripts/get-version-marker').replace(/\./g, '\.')")
+if find . -name package.json | grep -v node_modules | xargs grep -E "\"[^0-9]?${marker}\""; then
```
Comment (eladb) [inline scripts/align-version.sh:17]: Do we really need the `^[0-9]` now that you've added quotes?

Comment (jogold) [reply scripts/align-version.sh:17]: Maybe replace with `[\^~]?` if you want at some point to match caret or tilde versions?

Comment (eladb) [reply scripts/align-version.sh]: Yes, I think we still need this

**Comment by eladb** [comment]
@RomainMuller can you take another look?

**Pull request merged**